File: ggcommons/config/manager/environment_config_manager.py
import json
import logging
import os

from ggcommons.config.manager.config_manager import ConfigManager

logger = logging.getLogger("EnvironmentConfigManager")


class EnvironmentConfigManager(ConfigManager):
    def __init__(
        self, thing_name: str, component_name: str, environment_variable_name: str
    ):
        self._environment_variable_name = environment_variable_name
        super().__init__(component_name, thing_name)
        self._config_source = f"Environment (var name: {environment_variable_name})"
        self.init()

    def _load_configuration(self) -> dict:
        if self._environment_variable_name not in os.environ:
            logger.fatal(
                f"Expecting Greengrass component configuration in '{self._environment_variable_name}' environment variable. "
                f"Check component recipe to ensure '{self._environment_variable_name}' environment variable is set to the "
                f"component configuration in the 'Run' lifecycle section."
            )
            exit(1)
        return json.loads(os.environ.get(self._environment_variable_name))

    def get_config_source(self) -> str:
        return f"Environment (var name: {self._environment_variable_name})"

File: src/main/java/com/aws/proseve/ggcommons/config/manager/EnvironmentConfigManager.java
package com.aws.proseve.ggcommons.config.manager;

import com.github.cliftonlabs.json_simple.JsonException;
import com.github.cliftonlabs.json_simple.JsonObject;
import com.github.cliftonlabs.json_simple.Jsoner;

class EnvironmentConfigManager extends ConfigManager
{
    private final String environmentVariableName;

    EnvironmentConfigManager(String componentName, String environmentVariableName)
    {
        super(componentName);
        this.environmentVariableName = environmentVariableName;
        init();
    }

    @Override
    protected JsonObject loadConfiguration()
    {
        JsonObject retVal = null;

        LOGGER.debug("Loading configuration from environment variable '{}'", environmentVariableName);
        String configStr = System.getenv(environmentVariableName);
        if (configStr != null)
        {
            try
            {
                retVal = (JsonObject) Jsoner.deserialize(configStr);
            }
            catch (JsonException e)
            {
                LOGGER.fatal("Error parsing configuration: {}\n{}", configStr, e.toString());
                System.exit(1);
            }
        }
        else
        {
            LOGGER.fatal("Configuration environment variable ('{}') not defined.", environmentVariableName);
            System.exit(2);
        }

        return retVal;
    }

    @Override
    protected String getConfigSource()
    {
        return String.format("Environment (var name: %s)", environmentVariableName);
    }
}
